### PR TITLE
Preformatted: save line breaks as characters

### PR DESCRIPTION
**Pull request by ellatrix**
## Description
See #14564.

Preformatted is a special element: [the structure is represented by typographic conventions rather than by elements](https://html.spec.whatwg.org/multipage/grouping-content.html#the-pre-element). This means that character line breaks are displayed, and line break elements are safe to remove.

* It's easier to read the HTML version of the block, and it looks cleaner.
* It resolves problems for content processors that expect normal line breaks (even though element line breaks are valid as well).

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Review by gziolo — APPROVED** [review]
Looks good 👍

**Comment by ellatrix** [comment]
Thanks!

File: packages/e2e-tests/specs/__snapshots__/adding-blocks.test.js.snap [hunk 1]
```diff
@@ -42,7 +42,9 @@ exports[`adding blocks Should insert content using the placeholder and the regul
 <!-- /wp:quote -->
 
 <!-- wp:preformatted -->
-<pre class=\\"wp-block-preformatted\\">Pre text<br><br>Foo</pre>
+<pre class=\\"wp-block-preformatted\\">Pre text
+
```
Comment (aduth) [inline packages/e2e-tests/specs/__snapshots__/adding-blocks.test.js.snap]: Would this change invalidate existing content?

Comment (ellatrix) [reply packages/e2e-tests/specs/__snapshots__/adding-blocks.test.js.snap]: I’m not sure why it would? Only the onChange handler has changed

Comment (aduth) [reply packages/e2e-tests/specs/__snapshots__/adding-blocks.test.js.snap]: Ah, right. So I expect it should remain valid for the initial load. The first edit may convert some `<br>` to newlines, which is fine.

**Pull request merged**